File: src/2570.merge-two-2d-arrays-by-summing-values.2707/solution.ts
function mergeArrays (nums1: number[][], nums2: number[][]): number[][] {
    const result:number[][] = [];
    let index1 = 0;
    let index2 = 0;
    while (index1 < nums1.length && index2 < nums2.length) {
        if (nums1[index1][0] < nums2[index2][0]) {
            result.push(nums1[index1++]);
        } else if (nums1[index1][0] === nums2[index2][0]) {
            result.push([nums1[index1][0], nums1[index1++][1] + nums2[index2++][1], ]);
        } else {
            result.push(nums2[index2++]);
        }
    }
    while (index1 < nums1.length) {
        result.push(nums1[index1++]);
    }
    while (index2 < nums2.length) {
        result.push(nums2[index2++]);
    }
    return result;
}

File: src/2570.merge-two-2d-arrays-by-summing-values.2707/solution.java
class Solution {
    public int[][] mergeArrays(int[][] nums1, int[][] nums2) {
        List<int[]> result = new ArrayList<>();
        int i = 0;
        int j = 0;
        while (i < nums1.length && j < nums2.length) {
            if (nums1[i][0] < nums2[j][0]) {
                result.add(nums1[i++]);
            } else if (nums1[i][0] > nums2[j][0]) {
                result.add(nums2[j++]);
            } else {
                nums1[i][1] += nums2[j][1];
                result.add(nums1[i]);
                i++;
                j++;
            }
        }
        while (i < nums1.length) {
            result.add(nums1[i++]);
        }
        while (j < nums2.length) {
            result.add(nums2[j++]);
        }
        return result.toArray(new int[0][]);
    }
}